### PR TITLE
Manage asciidoctor with bundler

### PR DESCRIPTION
**Pull request by peaceiris**
I enhanced the workflow for fast execution!

Issue: https://github.com/peaceiris/actions-hugo/issues/342
ruby/setup-ruby: https://github.com/ruby/setup-ruby
bundle command: https://bundler.io/

Actions log (Completed in 12s) : https://github.com/peaceiris/today-i-learned/runs/711724694?check_suite_focus=true

### CODE REVIEW

File: .github/workflows/gh-pages.yml [hunk 1]
```diff
@@ -13,29 +13,29 @@ jobs:
         with:
           submodules: true
 
-          # Does not provide asciidoctor out of the box :(
```
Comment (reitzig) [inline .github/workflows/gh-pages.yml:19]: :grin:

Comment (peaceiris) [reply .github/workflows/gh-pages.yml:19]: Do you want to keep this line?

**Comment by reitzig** [comment]
Love the "magic" of the deploy action so that it tests the build but doesn't deploy on non-master branches! Didn't realize that was the case, thanks!

File: .github/workflows/gh-pages.yml (hunk 1 above)
Comment (peaceiris) [reply .github/workflows/gh-pages.yml:19]: ʕ◔ϖ◔ʔ 👍

**Comment by reitzig** [comment]
@peaceiris, what do you think of [actions-asciidoctor](https://github.com/reitzig/actions-asciidoctor) instead? :grimacing: 

Unfortunately, while that form of `asciidoctor` hack seems to work, [_here_](https://github.com/reitzig/today-i-learned/runs/719982901?check_suite_focus=true) it fails:

> fork/exec /home/runner/work/today-i-learned/today-i-learned/actions_asciidoctor/asciidoctor: exec format error

File: .github/workflows/gh-pages.yml [hunk 2]
```diff
@@ -13,29 +13,29 @@ jobs:
...
-          sudo gem install asciidoctor
-          sudo .github/scripts/reroute_asciidoctor.sh
-          ruby --version; gem --version; asciidoctor --version
+      - uses: ruby/setup-ruby@v1
+        with:
+          ruby-version: 2.7
+
+      - run: bundle config set path 'vendor/bundle'
+      - run: bundle install
+      - run: bash .github/scripts/reroute_asciidoctor.sh
```
Comment (peaceiris) [inline .github/workflows/gh-pages.yml]: BTW, I think we can just replace the script with the following alias.

```diff
- sudo .github/scripts/reroute_asciidoctor.sh
+ alias asciidoctor="asciidoctor --attribute=experimental=true --attribute=icons=font"
```

Or run it before running Hugo.

**Comment by peaceiris** [comment]
OK. It passed.

https://github.com/reitzig/today-i-learned/runs/720322514

**Comment by reitzig** [comment]
Good alternative, thanks!

I'll [stick to the action](https://github.com/reitzig/today-i-learned/pull/3) for now (and hope for future simplications regarding ruby setup).

Cheers!

**Pull request closed without merging**